core-rust/core-api-server/src/core_api/handlers/stream_transactions.rs

```diff
@@ -51,10 +51,27 @@ pub(crate) async fn handle_stream_transactions(
             Please note the resync will take a while.",
         ));
     }
+    let previous_state_identifiers = match from_state_version.previous() {
```

Comment:
Might be nice to have a mini integration test for this - eg stream 5 transactions, then another 5, and check the identifiers agree between both